Source/Core/Core/HW/DSPHLE/DSPHLE.cpp

```diff
@@ -191,26 +191,37 @@ void DSPHLE::DSP_WriteMailBoxLow(bool cpu_mailbox, u16 value)
 // Other DSP functions
 u16 DSPHLE::DSP_WriteControlRegister(u16 value)
 {
-  DSP::UDSPControl temp(value);
```

Comment:
I was thinking about that, but I looked through the rest of the DSPHLE code, and there's no in-between reads (or use of the control register at all, actually). This looks to me like an overly-literal translation of the DSPLLE code, which modifies the function argument directly, presumably because it's simpler than typing out `g_dsp.cr` each time:

```c++
// reset
if (val & 1)
{
  INFO_LOG(DSPLLE, "DSP_CONTROL RESET");
  DSPCore_Reset();
  val &= ~1;
}
// init
else if (val == 4)
{
  // HAX!
  // OSInitAudioSystem ucode should send this mail - not DSP core itself
  INFO_LOG(DSPLLE, "DSP_CONTROL INIT");
  g_init_hax = true;
  val |= 0x800;
}

// update cr
g_dsp.cr = val;
```

DSPLLE has no in-between uses of the control register either.

I guess my question is, is this an emulation accuracy thing, or just a quirk of the codebase?